.github/workflows/ci.yml

```diff
@@ -0,0 +1,14 @@
+name: CI
+on:
+  pull_request:
+  workflow_dispatch:
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    name: Deploy
+    steps:
+      - uses: actions/checkout@master
+      - name: Test
+        run: make test
+      - name: Preview
+        run: cd pulumi && pulumi preview
```

Comment:
nit: we can use pulumi/actions@v3 here and it will allow us to specify a working dir etc - https://github.com/pulumi/actions#getting-started but we can change to use that